### PR TITLE
test: Migrate some json tests to snapbox

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

This builds on assert-rs/snapbox#348 and is part of #14039.

Note: this also updates existing `.is_jsonlines()` usage to `.is_json().against_jsonlines()`.

### CODE REVIEW

File: tests/testsuite/test.rs [hunk 1]
```diff
@@ -4428,31 +4428,49 @@ fn json_artifact_includes_executable_for_library_tests() {
...
-            "#,
+        .with_stdout_data(
+            str![[r#"
+[
+  {
+    "executable": "[ROOT]/foo/target/debug/deps/foo-[HASH]",
+    "features": [],
+    "filenames": [
+      "[ROOT]/foo/target/debug/deps/foo-[HASH]"
+    ],
```
Comment (weihanglo) [inline tests/testsuite/test.rs:4439]: Omit `filenames` and `profile` like before?

**Review by weihanglo — COMMENTED** [review]
If this is the way to go, should we document how to assert JSON lines output?

**Comment by epage** [comment]
> If this is the way to go, should we document how to assert JSON lines output?

I've updated the deprecation message but further documentation will be in a followup documentation PR related to all of #14039.  This is prep for that.

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
@bors r+

Thank you.

**Pull request merged**